### DISCUSSION TITLE
regions[id].playLoop() often plays a region before that (same case with {loop:true} with play())

### DISCUSSION THREAD

**Discussion by logesh2496** [katspaugh/wavesurfer.js#2980] [Q&A]
Hi, I have multiple regions created and a normal playback(play()) of the regions works well. 

The problem occurs when using playLoop() or {loop: true) with play() method. It often plays one region before that instead of the actual region.

Example:

created regions: 1, 2, 3, 4
If I keep changing the playback of regions with playLoop(). it starts playing incorrect region (ofter 1 region before that: if I playLoop() region id 4 it will play 3)

**Comment by logesh2496** [comment]
New Observation: The issue will be reproduced only if we try to play consecutive regions. It always works if we play +2nd region.

Example: 
Region ids: 1, 2, 3, 4

It works: if we play region 1 then play 3 or 4. (1->3)
**_Doesn't work_**: if we play region 1 then try to play 2. (1->2)

**Comment by katspaugh** [comment]
Sounds like you're talking about v6? It won't be supported anymore. I suggest that look into implementing looping in your own app following the example here:

https://wavesurfer-js.org/examples/#regions.js

**Comment by logesh2496** [comment]
got it. yes, I'm using v6.6.4. 

which version is recommended?

**Reply by katspaugh** [reply]
v7
